# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/sandbox/chromium/base/task_runner.h
# Commit: 370c40e49230
# Full Hash: 370c40e4923073a87845b57c1f4851bacd9663da
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-12 21:42:35
# Regressor Bug: 1552160
# File Overlap Count: 3
# Description:
#   Bug 1552160 Part 1: Roll-up of chromium sandbox update and mozilla patches to get a running browser. r=jld,aklotz,tjr,bobowen
#   
#   This updates security/sandbox/chromium/ files to chromium commit 84108231f6e6e0772fb9a4643679ce76aa771e67.
#   
#   Existing and new patches applied from security/sandbox/chromium-shim/patches/with_update/ to give a compiling and mostly working browser.
# ==============================================================================

diff -r bd5ecd72f76c -r 370c40e49230 security/sandbox/chromium/base/task_runner.h
--- a/security/sandbox/chromium/base/task_runner.h	Wed Jun 12 21:55:59 2019 +1200
+++ b/security/sandbox/chromium/base/task_runner.h	Wed Jun 12 11:10:48 2019 +0100
@@ -61,12 +61,12 @@
   // will not be run.
   //
   // Equivalent to PostDelayedTask(from_here, task, 0).
-  bool PostTask(const tracked_objects::Location& from_here, OnceClosure task);
+  bool PostTask(const Location& from_here, OnceClosure task);
 
   // Like PostTask, but tries to run the posted task only after |delay_ms|
   // has passed. Implementations should use a tick clock, rather than wall-
   // clock time, to implement |delay|.
-  virtual bool PostDelayedTask(const tracked_objects::Location& from_here,
+  virtual bool PostDelayedTask(const Location& from_here,
                                OnceClosure task,
                                base::TimeDelta delay) = 0;
 
@@ -129,17 +129,13 @@
   //   * The DataLoader object can be deleted while |task| is still running,
   //     and the reply will cancel itself safely because it is bound to a
   //     WeakPtr<>.
-  bool PostTaskAndReply(const tracked_objects::Location& from_here,
+  bool PostTaskAndReply(const Location& from_here,
                         OnceClosure task,
                         OnceClosure reply);
 
  protected:
   friend struct TaskRunnerTraits;
 
-  // Only the Windows debug build seems to need this: see
-  // http://crbug.com/112250.
-  friend class RefCountedThreadSafe<TaskRunner, TaskRunnerTraits>;
-
   TaskRunner();
   virtual ~TaskRunner();
 